oe_contact_forms.module

```diff
@@ -36,17 +38,53 @@ function oe_contact_forms_form_alter(&$form, FormStateInterface $form_state, $fo
   if (str_starts_with($form_id, 'contact_message_')) {
     // Set the maximum length for the message field to 2000 characters.
     if (isset($form['message']['widget'][0]['value'])) {
-      $limit = 2000;
-      $form['message']['widget'][0]['value']['#maxlength'] = $limit;
+      $form['#validate'][] = 'oe_contact_forms_max_length_validate';
+      $form['message']['widget'][0]['value']['#attributes']['maxlength'] = OE_CONTACT_FORMS_MESSAGE_MAX_LENGTH;
       if (Drupal::service('module_handler')->moduleExists('maxlength')) {
         $form['message']['widget'][0]['value']['#attributes']['class'][] = 'maxlength';
-        $form['message']['widget'][0]['value']['#attributes']['data-maxlength'] = $limit;
+        $form['message']['widget'][0]['value']['#attributes']['data-maxlength'] = OE_CONTACT_FORMS_MESSAGE_MAX_LENGTH;
         $form['#attached']['library'][] = 'maxlength/maxlength';
       }
     }
   }
 }
 
+/**
+ * Validates the maximum length of the contact message.
+ *
+ * This function checks if the message exceeds the defined maximum length
+ * and sets an error if it does.
+ *
+ * @param array $element
+ *   The form element being validated.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current form state.
+ */
+function oe_contact_forms_max_length_validate(array &$element, FormStateInterface $form_state): void {
+  if (!isset($element['message']['widget'][0]['value'])) {
+    return;
+  }
+
+  $target =& $element['message']['widget'][0]['value'];
+  $value = (string) ($target['#value'] ?? '');
+  $max = OE_CONTACT_FORMS_MESSAGE_MAX_LENGTH;
+  // Fix for count one line as 2x chars.
```

Comment:
I don't think this comment is relevant because:
1. It's ambiguous.
2. Is not fixing anything.

Please adapt it so it reflects what happens here.